test/framework/modules.py

```diff
@@ -178,6 +178,18 @@ def test_load(self):
         for mod in mods:
             self.assertErrorRegex(EasyBuildError, '.*', self.modtool.load, [mod])
 
+        # by default, modules are always loaded, even if they are already loaded
+        self.modtool.load(['GCC/4.7.2'])
+        del os.environ['EBROOTGCC']
+        self.assertTrue('GCC/4.7.2' in self.modtool.loaded_modules())
+        self.modtool.load(['GCC/4.7.2'])
+        self.assertTrue(os.environ.get('EBROOTGCC'))
+
+        # reloading can be disabled using allow_reload=False
+        del os.environ['EBROOTGCC']
+        self.modtool.load(['GCC/4.7.2'], allow_reload=False)
+        self.assertEqual(os.environ.get('EBROOTGCC'), None)
+
```

Comment:
I am not sure I like the test. It does the job but if the target is to make sure the `$MODULEPATH` stays the same, doesn't it make more sense to check if it changed in anyway, rather than rely in a variable `$EBROOTXXX` that might or might not be set in the future (due to a bug, or changes in EB)?

Also, checking for changes in the `$MODULEPATH` won't be enough in flat module naming schemes, I guess the order of the loaded modules should be checked as well. But it seems pretty doable.